elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,414 @@
+        indent = int(block_indents[index])
+        answer_code += ('    ' * indent) + answer + '\n'
+    answer_code = leading_code + '\n' + answer_code + trailing_code + '\n'
+    return answer_code
+
+
+def prepare(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+
+    pl.check_attribs(element,
+                     required_attribs=['answers-name'],
+                     optional_attribs=['shuffle-source-blocks',
+                                       'grading-method',
+                                       'check-indentation',
+                                       'source-header',
+                                       'solution-header',
+                                       'file-name',
+                                       'leading-code',
+                                       'trailing-code',
+                                       'solution-placement',
+                                       'max-incorrect',
+                                       'min-incorrect',
+                                       'weight'])
+
+    answer_name = pl.get_string_attrib(element, 'answers-name')
+
+    mcq_options = []
+    html_ordering = []
+    correct_answers = []
+    correct_answers_indent = []
+    correct_answers_ranking = []
+    incorrect_answers = []
+
+    for html_tags in element:  # iterate through the tags inside pl-order-blocks, should be <pl-answer> tags
+        if html_tags.tag == 'pl-answer':
+            # correct attribute is not strictly required, as the attribute is irrelevant for autograded questions
+            pl.check_attribs(html_tags, required_attribs=[], optional_attribs=['correct', 'ranking', 'indent'])
+
+            isCorrect = pl.get_boolean_attrib(html_tags, 'correct', PL_ANSWER_DEFAULT_CORRECTNESS)
+            answerIndent = pl.get_string_attrib(html_tags, 'indent', PL_ANSWER_DEFAULT_INDENT)  # get answer indent, and default to -1 (indent level ignored)
+            if isCorrect is True:
+                # add option to the correct answer array, along with the correct required indent
+                if pl.get_string_attrib(html_tags, 'ranking', '') != '':
+                    ranking = pl.get_string_attrib(html_tags, 'ranking')
+                    try:
+                        ranking = int(ranking) - 1
+                    except ValueError:
+                        raise Exception('Ranking specified in <pl-answer> is not a number.')
+                    correct_answers_ranking.append(ranking)
+                correct_answers.append(html_tags.text)
+                correct_answers_indent.append(answerIndent)
+            else:
+                incorrect_answers.append(html_tags.text)
+            html_ordering.append(html_tags.text)
+        else:
+            raise Exception('Tags nested inside <pl-order-blocks> must be <pl-answers>.')
+
+    if pl.get_string_attrib(element, 'grading-method', 'ordered') != 'external' and len(correct_answers) == 0:
```

Comment:
Please use the `_DEFAULT` variables defined at the top of this file everywhere we are using a default (that is, don't use the hard-coded string `'ordered'` here).

Reply:
Also, we should check if the `grading-methods` are one of the defined methods. I just noticed one of the examples still had `any`, but the question rendered without any errors (however, not the correct behavior).

Reply:
added and fixed